packages/helix-shared-config/src/parsers/BaseParser.js

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright 2021 Adobe. All rights reserved.
+ * This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+
+class BaseParser {
```

Comment:
IMO there is no need for creating an extra base parser, since we don't have another kind (yet).
Following KISS, I would add a one, once there is a need for a abstraction.

Reply:
I was planning on moving the YAML stuff into a separate parser in a separate PR, right now the YAML document is coupled to the config. I decided to not make any changes yet since it would mean a big PR. 

Happy to drop the base and skip that if that's preferred, wdyt?

Reply:
good idea, then keep it this way.